lib/opts/index.ts

```diff
@@ -0,0 +1,122 @@
+type Hook =
+  | "prerelease"
+  | "prebump"
+  | "postbump"
+  | "prechangelog"
+  | "postchangelog"
+  | "precommit"
+  | "postcommit"
+  | "pretag"
+  | "posttag";
+
+type TypePrefixes = Array<
+  (
+    | { section: string; hidden?: boolean | undefined }
+    | { hidden: true; section?: string | undefined }
+  ) & { type: string }
+>;
+
+type ConfigFiles = Array<
+  | string
+  | { filename: string; type: "json" | "gradle" | "plain-text" }
+  | { filename: string; updater: string }
+>;
+
+/**
+ * __THIS SHOULD NOT CHANGE.__
+ * @deprecated
+ *
+ * The configuration options for `standard-version` as of version 9.5 (The last version prior to the fork; deprecated).
+ */
+export type SVConfig = {
+  packageFiles: ConfigFiles;
+  bumpFiles: ConfigFiles;
+  releaseAs: Release;
+  prerelease: string | boolean;
+  infile: string;
+  message: string;
+  firstRelease: boolean;
+  sign: boolean;
+  noVerify: boolean;
+  commitAll: boolean;
+  silent: boolean;
+  tagPrefix: string;
+  scripts: Record<Hook, string>;
+  skip: Record<Task, string>;
+  dryRun: boolean;
+  gitTagFallback: boolean;
+  path: string;
+  changelogHeader: string;
+  preset: string;
+  lernaPackage: string;
+  header: string;
+  types: TypePrefixes;
+  preMajor: boolean;
+  commitUrlFormat: string;
+  compareUrlFormat: string;
+  issueUrlFormat: string;
+  userUrlFormat: string;
+  releaseCommitMessageFormat: string;
+  issuePrefixes: string[];
+};
+
+/**
+ * The configuration object for commit-and-tag-version, which is a superset of the conventional-changelog-config-spec (as of version 2.1.0)
+ * This may or may not maintain backwards compatibility with standard-version (as of version 9.5.0).
+ */
+export type CatVConfig = PrettyPrint<
+  SVConfig & {
+    npmPublishHint: string;
+    releaseCount: number;
+    tagForce: boolean;
+  }
+>;
+
+/** The configuration options that are not supported by standard-version (as of version 9.5.0). */
+const catVOnlyFeatures = [
+  "npmPublishHint",
+  "releaseCount",
+  "tagForce",
+] as const satisfies ReadonlyArray<Exclude<keyof CatVConfig, keyof SVConfig>>;
+
+export const getMergedConfig = async (
+  cwd?: string
+): Promise<Partial<CatVConfig>> => {
+  const dir = cwd ?? process.cwd();
+  const json = (await import("path")).join(dir, "package.json");
+  const svConfig: SVConfig = (await import(json))["standard-version"];
+  const catvConfig: CatVConfig = (await import(json))["commit-and-tag-version"];
+
+  const config: Partial<CatVConfig> = {};
+
+  if (svConfig) {
+    for (const key in svConfig) {
+      // @ts-expect-error - We know this key SHOULD NOT be in svConfig, but we're going to allow it and warn the user
+      if (catVOnlyFeatures.includes(key)) {
+        console.warn(
+          `The "${key}" option is a feature of commit-and-tag-version, and is not supported by standard-version.${"\n"}Please move this option to the 'commit-and-tag-version' key.${"\n"}In a future version, this will throw an error.`
```

Comment:
What's the reason this should be an error / warning?

Reply:
This was more of a style choice. River I figured if the API is going to diverge, we might consider retroactively making the point of divergence the 'fork-point', so to speak - allowing us to freeze in time `standard-version` compatibility, and force further changes to consider the legacy options (by providing some sort of shim).

Reply:
I like your philosophy of making it easy for users to migrate and warning them about what's going on.

I'd assumed we'd just support both keys the same way, but I don't feel strongly, so happy to go with whatever you prefer. I don't think this case will come up much, so it's not worth spending a great deal of time on the perfect implementation.

I suppose the best compatibility would be for us to eventually remove the `standard-version` key - so if (for whatever reason) someone wanted both tools, they could without conflict. Again, super rare and unlikely.

I'm happy with whatever you think is right here, as long as it doesn't introduce maintenance overhead.

Reply:
As long as API changes are *additive* to the SV API, the burden is 0. If there is a non-backwards compatible breaking change, a shim would need to be developed or the standard-version key deprecated.